File: BloodFlow360.Domain/Entities/Donor.cs
using BloodFlow360.Domain.Common;

namespace BloodFlow360.Domain.Entities;

public class Donor : BaseEntity
{
    public Guid UserId { get; set; }

    public User User { get; set; } = null!;

    public Guid BloodBankId { get; set; }

    public BloodBank BloodBank { get; set; } = null!;

    public string DonorNumber { get; set; } = string.Empty;

    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public DateOnly DateOfBirth { get; set; }

    public string Gender { get; set; } = string.Empty;

    public string BloodGroup { get; set; } = string.Empty;

    public decimal Height { get; set; }

    public decimal Weight { get; set; }

    public string AadhaarNumber { get; set; } = string.Empty;

    public string Occupation { get; set; } = string.Empty;

    public string EmergencyContactName { get; set; } = string.Empty;

    public string EmergencyContactNumber { get; set; } = string.Empty;

    public DateOnly? LastDonationDate { get; set; }

    public bool IsEligible { get; set; }

    public bool IsActive { get; set; }
}

File: BloodFlow360.React/src/types/donor.ts
export interface Donor {
  id: string;
  userId: string;
  bloodBankId: string;
  donorNumber: string;
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  gender: string;
  bloodGroup: string;
  height: number;
  weight: number;
  aadhaarNumber: string;
  occupation: string;
  emergencyContactName: string;
  emergencyContactNumber: string;
  lastDonationDate: string | null;
  isEligible: boolean;
  isActive: boolean;
  createdAt: string;
}

export interface CreateDonor {
  userId?: string;
  bloodBankId: string;
  donorNumber: string;
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  gender: string;
  bloodGroup: string;
  height: number;
  weight: number;
  aadhaarNumber: string;
  occupation: string;
  emergencyContactName: string;
  emergencyContactNumber: string;
  isEligible: boolean;
  isActive: boolean;
}

export interface UpdateDonor {
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  gender: string;
  bloodGroup: string;
  height: number;
  weight: number;
  aadhaarNumber: string;
  occupation: string;
  emergencyContactName: string;
  emergencyContactNumber: string;
  isEligible: boolean;
  isActive: boolean;
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
  errors?: string[];
}

export interface PagedResponse<T> extends ApiResponse<T> {
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
}

export const BLOOD_GROUPS = [
  "A+", "A-", "B+", "B-", "AB+", "AB-", "O+", "O-",
] as const;

export const GENDERS = ["Male", "Female", "Other"] as const;